garage/logger/singleton_logger.py

```diff
@@ -0,0 +1,99 @@
+"""This is Garage's logger singleton.
```

Comment:
Can you provide a summary here of the principles of operation.

Reply:
@krzentner @ryanjulian What do you mean by "principles of operation?"